client.py

```diff
@@ -146,25 +149,71 @@ async def handle_add_game_request(message, game_name):
         reply = 'This game doesn\'t exist on steam, try gamepass.'
     await message.reply (reply)
 
+async def handle_delete_game_request_from_match(message, index_string):
+    index = int(index_string) - 1
+    game_matches = client_state.get_game_matches()
+    if len(game_matches) == 0:
+        print('game matches is empty')
+        return
+    game_tupple = game_matches[index]
+    deleted_game_result = delete_game(game_tupple)
+    print('Deleted game result is:', deleted_game_result)
+    if deleted_game_result == True:
+        reply = f'**{game_tupple[1]}** was successfully deleted from your wishlist.'
+    await message.reply(reply)
+
 # Function to handle client side of a delete game request, builds reply with list_games_for_reply function
-async def handle_delete_game_request(message, entries):
+async def handle_delete_game_request(message, game_name):
     reply = ''
-    if len(entries) == 1:
-        entry = entries[0]
-        status = delete_game((entry["appid"], entry["name"]))
+    matches = get_entries_from_wishlist(game_name)
+    if len(matches) == 1:
+        # Working with a list with a single tupple in it from matches
+        tupple = (matches[0][0], matches[0][1])
+        status = delete_game(tupple)
+        game_title = matches[0][1]
+        client_state.clear_game_matches()
         if status == True:
-            reply = f'**{entry["name"]}** was successfully deleted from our tracking list.'
+            reply = f'**{game_title}** was successfully deleted from your wishlist.'
+        else: # Not exact match
+            reply = f'Did you mean **{game_title}**? Write "Yes" to confirm'
+            client_state.store_game_matches(matches)
+    elif len(matches) >1:
+        # working with a list with multiple tupples in it, iterate through list of tupples which have this format (appid, app name)
+        reply = 'You are not tracking a game with that exact title, did you mean one of these?\n'
+        store_matches = True
+        for index, app in enumerate(matches):
+            game_title = app[1]
+            reply += f'\n{index + 1}: {game_title}'
+            print('app is:', app)
+            if len(reply) > 2000: 
+                reply = f'There are {len(matches)} apps that contain that title, can you be more specific?'
+                store_matches = False
+                break
+        if store_matches:
+            client_state.store_game_matches(matches)
+            reply = reply + f'\n\nType "delete #" to delete the specific title. We\'ll keep track of this list for 2 minutes.'
     else:
-        output = games_were_tracking_string()
-        reply = f'There was a problem deleting the game, are we tracking it?\n{output}'
+        reply = 'This game doesn\'t exist on steam, try gamepass.'

```

Comment:
Could this be confusing since we're trying to delete?

Reply:
yup, resolved